### ISSUE TITLE
Clang doesn't issue any variable alias information

### ISSUE THREAD

**Issue by billionai** [llvm/llvm-project#52664]
Following [this tutorial](https://www.keil.com/support/man/docs/armclang_ref/armclang_ref_chr1385462324931.htm) and checking both, the disassembly of the result (with objdump -d) and the DWARF information (with readelf -w), doesn't seem to yield any information on newname. 

A more minimal reproducer would be:

```
int old = 1;
extern int newvar __attribute__ ((alias("old")));
int main(){
    newvatr = 0;
    return old;
}
```

This could make debugging programs that use aliases much more complicated to debug, since GDB won't be able to resolve these aliases to their original values without DWARF information on these aliases

**Comment by dwblaikie** [comment]
Yep, seems fair. Can confirm that Clang/LLVM doesn't produce a description of `newvar` but GCC does. GCC's description is:
```
0x0000004a:   DW_TAG_variable
                DW_AT_name      ("newvar")
                DW_AT_decl_file ("/usr/local/google/home/blaikie/dev/scratch/test.cpp")
                DW_AT_decl_line (2)
                DW_AT_decl_column       (0x0c)
                DW_AT_type      (0x00000043 "int")
                DW_AT_external  (true)
```
